src/app/ui/main_menu_bar.cpp

```diff
@@ -26,13 +28,23 @@ MainMenuBar::MainMenuBar()
 
   // Reload the main menu if there are changes in keyboard shortcuts
   // or scripts when extensions are installed/uninstalled or
-  // enabled/disabled.
-  m_extKeys = extensions.KeysChange.connect([this] { reload(); });
-  m_extScripts = extensions.ScriptsChange.connect([this] { reload(); });
+  // enabled/disabled, enqueued to avoid multiple reloads.
+  auto reloadCbk = [this] {
+    if (m_queuedReload)
+      return;
+    m_queuedReload = true;
+
+    auto* msg = new CallbackMessage([this] { reload(); });
+    msg->setRecipient(this);
+    Manager::getDefault()->enqueueMessage(msg);
+  };
+  m_extKeys = extensions.KeysChange.connect(reloadCbk);
+  m_extScripts = extensions.ScriptsChange.connect(reloadCbk);
```

Comment:
Probably I'd move this `reloadCbk` code into a `delayedReload()` member function.